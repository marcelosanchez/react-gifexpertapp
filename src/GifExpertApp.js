import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {

    // const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
    const [categories, setCategories] = useState(defaultCategories)

    // const handleAdd = () => {
    //     setCategories( categories => [...categories, 'Kaladin'] )
    //     // console.log( categories )
    // };

    return <>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories = { setCategories }/>
        <hr />

        <ol>
            {
                categories.map( category => 
                        // <li key={ category }>{ category }</li>
                        <GifGrid 
                            key={ category } 
                            category={ category }
                        />
                    )
            }
        </ol>
    </>;
}

export default GifExpertApp;