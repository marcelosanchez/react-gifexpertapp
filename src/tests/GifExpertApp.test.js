import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en <GifExpertApp/>', () => {
    
    test('Debe mostrarse correctamente', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();
    });
    
    test('Debe mostrar una lista de categorias', () => {
        const categories = ['Puppies', 'birds'];
        const wrapper = shallow(<GifExpertApp defaultCategories = {categories}/>);
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );  // El numero de elementos debe ser igual al numero de categorias enviadas
    });
    
});