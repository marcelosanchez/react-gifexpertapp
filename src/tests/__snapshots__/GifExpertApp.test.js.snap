// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp/> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Puppies"
      key="Puppies"
    />
    <GifGrid
      category="birds"
      key="birds"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp/> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
